File: sai-backend/src/main/java/com/sai/backend/entity/Customer.java
package com.sai.backend.entity;

import jakarta.persistence.*;
import jakarta.validation.constraints.*;
import lombok.Data;
import lombok.EqualsAndHashCode;

@Data
@EqualsAndHashCode(callSuper = true)
@Entity
@Table(name = "customers")
public class Customer extends BaseEntity {
    
    @NotBlank(message = "客户名称不能为空")
    @Size(max = 100, message = "客户名称长度不能超过100个字符")
    @Column(name = "customer_name", nullable = false, length = 100)
    private String customerName;
    
    @Column(name = "status", nullable = false)
    private Boolean status = true; // true=启用，false=禁用
    
    @Size(max = 50, message = "国家长度不能超过50个字符")
    @Column(name = "country", length = 50)
    private String country;
    
    @Size(max = 50, message = "省份长度不能超过50个字符")
    @Column(name = "province", length = 50)
    private String province;
    
    @Size(max = 50, message = "城市长度不能超过50个字符")
    @Column(name = "city", length = 50)
    private String city;
    
    @Size(max = 50, message = "区县长度不能超过50个字符")
    @Column(name = "district", length = 50)
    private String district;
    
    @Size(max = 50, message = "联系人长度不能超过50个字符")
    @Column(name = "contact_person", length = 50)
    private String contactPerson;
    
    @NotBlank(message = "联系电话不能为空")
    @Pattern(regexp = "^1[3-9]\\d{9}$", message = "请输入正确的手机号码")
    @Column(name = "contact_phone", nullable = false, length = 20)
    private String contactPhone;
    
    @Email(message = "请输入正确的邮箱地址")
    @Size(max = 100, message = "邮箱长度不能超过100个字符")
    @Column(name = "contact_email", length = 100)
    private String contactEmail;
    
    @Column(name = "remark", columnDefinition = "TEXT")
    private String remark;
}

File: sai-frontend/src/services/customer.ts
import request from 'utils/request';

// 客户接口 - 与后端Customer实体保持一致
export interface ICustomer {
  id: string;
  customerName: string;  // 后端字段名
  status: boolean;
  country: string;
  province: string;
  city: string;
  district: string;
  contactPerson: string;
  contactPhone: string;
  contactEmail: string;
  remark: string;
  createTime: string;
  updateTime: string;
}

// 客户列表查询参数
export interface ICustomerListParams {
  pageSize: number;
  current: number;
  name?: string;     // 对应后端的客户名称搜索
  country?: string;  // 国家搜索参数
}

// 新增客户参数
export interface ICustomerCreateParams {
  customerName: string;
  status: boolean;
  country: string;
  province: string;
  city: string;
  district: string;
  contactPerson: string;
  contactPhone: string;
  contactEmail: string;
  remark: string;
}

// 获取客户列表 - 调用真实API
export const getCustomerList = async (params: ICustomerListParams) => {
  const response = await request.get('/customers', {
    params: {
      pageSize: params.pageSize,
      current: params.current,
      name: params.name,
      country: params.country,
    },
  });
  
  return {
    list: response.data.list,
    total: response.data.total,
  };
};

// 新增客户
export const createCustomer = async (params: ICustomerCreateParams) => {
  const response = await request.post('/customers', params);
  return response.data;
};

// 删除客户
export const deleteCustomer = async (id: string) => {
  const response = await request.delete(`/customers/${id}`);
  return response.data;
};

// 获取客户详情
export const getCustomerDetail = async (id: string) => {
  const response = await request.get(`/customers/${id}`);
  return response.data;
};

// 更新客户
export const updateCustomer = async (id: string, params: ICustomerCreateParams) => {
  const response = await request.put(`/customers/${id}`, params);
  return response.data;
};